### PR TITLE
Fixed issue reading issuer and subject from cert in diagnostic.  Issue:

### PR DESCRIPTION
**Pull request by sfc-gh-cconner**
https://github.com/snowflakedb/snowflake-connector-python/issues/1536

Also capture any exception during diagnostics and raise them after diagnostics complete

Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes [#NNNN](https://github.com/snowflakedb/snowflake-connector-python/issues/1536)

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ * ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

  We added a method that decrypts any bytes objects in cert dictionary.
  We capture all exceptions during diag test in a dictionary and raise it when the diag tests are done.

### CODE REVIEW

**Review by sfc-gh-sfan — COMMENTED** [review]
It might be hard to do integration test, but can we add some unit test such that we see exception being raised?

I don't have full context to know if an exception should be raised, but this seems reasonable to me. Please also add this to `DESCRIPTION` as part of changelog.

**Comment by sfc-gh-cconner** [comment]
> It might be hard to do integration test, but can we add some unit test such that we see exception being raised?
> 
> I don't have full context to know if an exception should be raised, but this seems reasonable to me. Please also add this to `DESCRIPTION` as part of changelog.

I have updated the 'DESCRIPTION' as part of the changelog.  As for an exception, they were all intentionally hidden and now with this change, they should bubble up.  However, since this fix fixes any Exception, I'm not quite sure how we would add a unit test to confirm an Exception is raised?

**Comment by sfc-gh-cconner** [comment]
@sfc-gh-mkeller , Hey, can you help me get this PR done?  Thanks!

File: test/unit/test_diag.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import snowflake.connector
+
+
+def test_exception_raise_during_diag_fail(monkeypatch, caplog):
```
Comment (sfc-gh-aalam) [inline test/unit/test_diag.py:11]: Could you also rename this file to `test_connection_diagnostic.py`?

**Comment by sfc-gh-sfan** [comment]
Test PR in https://github.com/snowflakedb/snowflake-connector-python/pull/1756

**Comment by lyderichti59** [comment]
Hello, what is the current progress on this issue ? I'm facing OSCP errors that I'd like to debug and a bug fix would be of great help. Seems like a PR is ready to be merged, anything missing to close this ?

**Comment by sfc-gh-cconner** [comment]
We just need to close this one out, the fixes merged in 3.3.0 or 3.2.1, can't remember which.  I see them in 3.5.0 for sure.  Have you tried with 3.5.0?

**Comment by sfc-gh-dszmolka** [comment]
per latest comments, this is implemented and released, closing it out.

**Pull request closed without merging**